pages/posts.js

```diff
@@ -1,41 +1,41 @@
-import getPosts from '../src/Services/get-posts';
+import { getPosts } from '../src/Services/get-posts';
 import Form from '../src/Components/Form/Form';
-import { saveToLocal, loadFromLocal } from '../src/Components/lib/localStorage';
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 import FormWrapper from '../src/Components/UI/Form/FormWrapper.styles';
-import dynamic from 'next/dynamic';
+
 import ButtonAdd from '../src/Components/UI/Form/Button/AddButton.styles';
+import Icon from '../src/Components/UI/Icons/icons';
+import { SWRConfig } from 'swr';
+import { swrFetcher } from '../src/Components/lib/swr-Fetcher';
+import RenderPosts from '../src/Components/Form/RenderPosts';
 
-export function getStaticProps() {
-	const data = getPosts();
+export async function getStaticProps() {
+	const posts = await getPosts();
+	console.log('posts: ' + JSON.stringify(posts));
 
 	return {
 		props: {
-			data,
+			fallback: {
+				'/api/posts': posts,
+			},
 		},
 	};
 }
-export default function Posts({ data }) {
-	const [posts, setPosts] = useState(loadFromLocal('localPosts') ?? data);
-	const [isFormActive, setIsFormActive] = useState(false);
 
-	const RenderPosts = dynamic(() => import('../src/Components/Form/RenderPosts'), {
-		ssr: false,
-	});
-
-	useEffect(() => {
-		saveToLocal('localPosts', posts);
-	}, [posts]);
+export default function Posts({ fallback }) {
+	const [isFormActive, setIsFormActive] = useState(false);
 
 	return (
 		<main>
-			<ButtonAdd type="button" onClick={() => setIsFormActive(prevCheck => !prevCheck)}>
-				Add new Post
-			</ButtonAdd>
-			<FormWrapper isFormActive={isFormActive}>
-				<Form posts={posts} onSetPosts={setPosts} onSetIsFormActive={setIsFormActive} />
-			</FormWrapper>
-			<RenderPosts posts={posts} onSetPosts={setPosts} />
+			<SWRConfig value={{ fetcher: swrFetcher, fallback }}>
+				<ButtonAdd type="button" onClick={() => setIsFormActive(prevCheck => !prevCheck)}>
+					<Icon variant="plus" />
+				</ButtonAdd>
+				<FormWrapper isFormActive={isFormActive}>
+					<Form onSetIsFormActive={setIsFormActive} />
+				</FormWrapper>
```

Comment:
Normally we want to keep the logic for setting states in the page file and not let the component do the work. I guess its better when you create a little function that automatically sets the state value here in the page component and pass this function down to the Form.

Reply:
Here you pass down the `setIsFormActive` function directly into your `Form`. This is something you should avoid.

Instead do something like this:

```
function deactivateForm() {
setIsFormActive(!isFormActive);
}
```

An then pass it down like this:

`<Form onSetIsFormActive={deactivateForm} />`

This way you don't separate the logic from the useState, means: You can see which changes to the useState can be applied without leaving the place where the useState was written.

This procedure is very important and is called "Lifting state up". Here is a [useful Link](https://reactwithhooks.netlify.app/docs/lifting-state-up.html)